src/sync/pull.ts

```diff
@@ -101,29 +100,46 @@ export async function beginPull(
     };
   }
 
+  const syncHead = await handlePullResponse(lc, store, baseCookie, response);
+  if (syncHead === null) {
+    throw new Error('Overlapping sync JsLogInfo');
+  }
+  return {
+    httpRequestInfo,
+    syncHead,
+  };
+}
+
+// Returns new sync head, or null if response did not apply due to mismatched cookie.
+export async function handlePullResponse(
+  lc: LogContext,
+  store: dag.Store,
+  expectedBaseCookie: ReadonlyJSONValue,
+  response: PullResponse,
+): Promise<Hash | null> {
   // It is possible that another sync completed while we were pulling. Ensure
   // that is not the case by re-checking the base snapshot.
   return await store.withWrite(async dagWrite => {
     const dagRead = dagWrite;
-    const mainHeadPostPull = await dagRead.getHead(db.DEFAULT_HEAD_NAME);
+    const mainHead = await dagRead.getHead(db.DEFAULT_HEAD_NAME);
 
-    if (mainHeadPostPull === undefined) {
+    if (mainHead === undefined) {
       throw new Error('Main head disappeared');
     }
-    const baseSnapshotPostPull = await db.Commit.baseSnapshot(
-      mainHeadPostPull,
-      dagRead,
-    );
-    if (baseSnapshot.chunk.hash !== baseSnapshotPostPull.chunk.hash) {
-      throw new Error('Overlapping syncs JSLogInfo');
+    const baseSnapshot = await db.Commit.baseSnapshot(mainHead, dagRead);
+    const [baseLastMutationID, baseCookie] =
+      db.Commit.snapshotMetaParts(baseSnapshot);
+
+    if (!deepEqual(expectedBaseCookie, baseCookie)) {
```

Comment:
I think this condition is weaker than what we had before, which was a check that the hash of the chunk didn't change. This checks that the cookie didn't change. Is it the case that two snapshots are exactly identical if and only if their cookies are identical? 
- is it not possible for users to not use cookies, in which case this check always passes even if snapshot changes
- i think it *is* possible at least theoretically that the LMID changes without the cookie changing, which makes this check incorrect I would think

Not sure about other cases? The original intent was "did the base snapshot change" and here we are using "is the cookie the same", which I'm not sure is saying the same thing.

Reply:
It's not the same. I am not 100% that the code as written makes sense for pull. But I think I am about to advocate for ditching pull from the protocol so perhaps it doesn't matter?